src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1748,6 +1752,21 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                         ImmutableArray.Create(new TypedConstant(compilation.GetWellKnownType(WellKnownType.System_Type), TypedConstantKind.Type, originalType)),
                         isOptionalUse: true));
             }
+
+            if (this.IsMicrosoftCodeAnalysisEmbeddedAttribute() && GetEarlyDecodedWellKnownAttributeData() is null or { HasCodeAnalysisEmbeddedAttribute: false })
```

Comment:
>&& GetEarlyDecodedWellKnownAttributeData() is null or { HasCodeAnalysisEmbeddedAttribute: false } [](http://example.com/codeflow?start=64&length=97)

Are we testing positive and negative effect of this condition?

Reply:
`EmbeddedAttributeFromSourceValidation_MissingEmbeddedAttributeApplicationGeneratesAttributeApplication` is the positive condition. There are a number of tests that cover the negative condition, where `Embedded` is applied to the attribute itself. Most directly, `EmbeddedAttributeFromSourceValidation_Valid`. However, I'll add a test with an unrelated early well-known attribute, to test the other positive condition.